src/Microsoft.Framework.Runtime.Roslyn/RoslynCompiler.cs

```diff
@@ -213,18 +213,59 @@ private CompilationModules GetCompileModules(ILibraryKey target)
         private static CSharpCompilation ApplyVersionInfo(CSharpCompilation compilation, Project project,
             CSharpParseOptions parseOptions)
         {
+            const string assemblyFileVersionName = "System.Reflection.AssemblyFileVersionAttribute";
+            const string assemblyVersionName = "System.Reflection.AssemblyVersion";
+            const string assemblyInformationalVersion = "System.Reflection.AssemblyInformationalVersion";
+
             var emptyVersion = new Version(0, 0, 0, 0);
+            var assemblyAttributes = compilation.Assembly.GetAttributes();
+
+            bool foundAssemblyFileVersion = false;
+            bool foundAssemblyVersion = false;
+            bool foundAssemblyInformationalVersion = false;
 
-            // If the assembly version is empty then set the version
-            if (compilation.Assembly.Identity.Version == emptyVersion)
+            foreach(var assembly in assemblyAttributes)
             {
-                return compilation.AddSyntaxTrees(new[]
+                string attributeName = assembly.AttributeClass.ToString();
+
+                if (attributeName == assemblyFileVersionName)
```

Comment:
> Maybe do  ... 

Ok

> Also is the value guaranteed to be an fully qualified name all the time? Even if the actual code didn't fully qualify it?

Good question and the answer is yes, it is always the full qualified name.